domain/src/main/java/co/nimblehq/domain/usecase/base/SingleUseCase.kt

```diff
@@ -0,0 +1,29 @@
+package co.nimblehq.domain.usecase.base
+
+import co.nimblehq.domain.data.error.AppError
+import io.reactivex.Scheduler
+import io.reactivex.Single
+
+abstract class SingleUseCase<in UseCaseInput, T>(
+    private val executionThread: Scheduler,
+    private val postExecutionThread: Scheduler,
+    defaultError: (Throwable) -> AppError
+) : BaseUseCase(defaultError) {
+
+    protected abstract fun create(input: UseCaseInput): Single<T>
+
+    fun execute(input: UseCaseInput): Single<T> {
+        return create(input)
+            .onErrorResumeNext {
+                if (isCanceledException(it)) {
+                    Single.never()
+                } else {
+                    Single.error(composeError(it))
+                }
+            }
+            .doOnError(::doOnError)
+            .subscribeOn(executionThread)
+            .observeOn(postExecutionThread)
+    }
+
```

Comment:
I think this is a redundant line, I've seen some end of the class has no empty line like this (such as [this line).](https://github.com/nimblehq/android-templates/pull/56/files#diff-1bb3f60c45467134a31713a8de5b38ddf45f8e8519e741ee9be96acf3bfebb10R19) What is our convention then?

Reply:
Correct this is a redundant line, applying detekt could resolve this but we could improve this later I think @lydiasama @manh-t